host/xtest/regression_1000.c

```diff
@@ -1010,6 +1072,119 @@ static void xtest_tee_test_1012(ADBG_Case_t *c)
 
 		TEEC_CloseSession(&session1);
 	}
+	Do_ADBG_EndSubCase(c, "Single Instance Multi Session");
+
+	Do_ADBG_BeginSubCase(c, "Multiple Instances Single Session");
+	{
+		uuid = miss_test_ta_uuid;
+
+		(void)ADBG_EXPECT_TEEC_SUCCESS(c, open_session(&cs[0], uuid));
+		(void)ADBG_EXPECT_TEEC_SUCCESS(c, open_session(&cs[1], uuid));
+
+		(void)ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+					do_get_counter(&cs[0], &counter));
+		(void)ADBG_EXPECT(c, 0, counter);
+
+		(void)ADBG_EXPECT_TEEC_RESULT(c, 0,
+					do_get_counter(&cs[1], &counter));
+		(void)ADBG_EXPECT(c, 0, counter);
+
+		(void)ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
+					do_panic(&cs[1]));
+		(void)ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+					do_get_counter(&cs[0], &counter));
+		(void)ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
+					do_get_counter(&cs[1], &counter));
+		close_session(&cs[1]);
+
+		(void)ADBG_EXPECT_TEEC_SUCCESS(c, open_session(&cs[2], uuid));
+
+		(void)ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+					do_get_counter(&cs[0], &counter));
+		(void)ADBG_EXPECT(c, 0, counter);
+		(void)ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
+					do_get_counter(&cs[2], &counter));
+		(void)ADBG_EXPECT(c, 0, counter);
+
+		close_session(&cs[0]);
+		close_session(&cs[2]);
```

Comment:
Can you factorize the sequence, i.e.:

```C
/* Return 0 on success and a non null value otherwise */
static int test_ta_panic_behavior(TEE_UUID *uuid, bool multi_instance)
{
	TEEC_Result exp_res = TEEC_ERROR_GENERIC;
	uint32_t counter = 0;
	uint32_t exp_counter = 0;
	struct xtest_session cs[3];

	memset(cs, 0, sizeof(cs));

	if (!ADBG_EXPECT_TEEC_SUCCESS(c, do_open_session(&cs[0], uuid)))
		goto bail;

	if (!ADBG_EXPECT_TEEC_SUCCESS(c, do_open_session(&cs[1], uuid)))
		goto bail0;

	if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
					do_get_counter(&cs[0], &counter)))
		goto bail1;

	if (!ADBG_EXPECT(c, 0, counter))
		goto bail1;

	if (!ADBG_EXPECT_TEEC_RESULT(c, 0,
					do_get_counter(&cs[1], &counter)))
		goto bail1;

	exp_counter = multi_instance ? 0 : 1;
	if (ADBG_EXPECT(c, 1, counter))
		goto bail1;

	if (!ADBG_EXPECT_TEEC_RESULT(c, ,
					do_panic(&cs[1])))
		goto bail1;

	exp_res = multi_instance ? TEEC_SUCCESS : TEE_ERROR_TARGET_DEAD;

	if (!ADBG_EXPECT_TEEC_RESULT(c, exp_res,
					do_get_counter(&cs[0], &counter)))
		goto bail1;

	if (!ADBG_EXPECT_TEEC_RESULT(c, TEE_ERROR_TARGET_DEAD,
					do_get_counter(&cs[1], &counter)))
		goto bail1;

		/* Open a new session on a new context */
	if (!ADBG_EXPECT_TEEC_SUCCESS(c, do_open_session(&cs[2], uuid)))
		goto bail1;

	/* Optional test, already tested above */
	if (multi_instance) {
		if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
						do_get_counter(&cs[0], &counter)))
			goto bail2;

		if (!ADBG_EXPECT(c, 0, counter))
			goto bail2;
	}

	if (!ADBG_EXPECT_TEEC_RESULT(c, TEEC_SUCCESS,
					do_get_counter(&cs[2], &counter)))
		goto bail2;

	if (!ADBG_EXPECT(c, 0, counter))
		goto bail2;

	ret = 0;

bail2:
	close_session(&cs[2]);
bail1:
	close_session(&cs[1]);
bail0:
	close_session(&cs[0]);
bail:
	return ret;
}
```

Reply:
Alright, this is a major improvement that avoid test code duplication. The new version include it as well.